lib/jarvis/commands/publish.rb

```diff
@@ -69,29 +69,36 @@ def execute
       if check_build?
         workdir_sha1 = git.revparse("HEAD")
         context[:sha1] = workdir_sha1
-        statuses = Octokit.statuses(project.path, workdir_sha1)
 
-        current_states = Hash[statuses.group_by(&:context).map do |ctx, info|
-           last_info = info.sort_by(&:created_at).last
-           [last_info.context, last_info.state]
-        end]
-
-        if current_states.any?
-          unsuccessful_statuses = Hash[current_states.select {|context,state| state != "success" }]
-          if unsuccessful_statuses.any?
-            logger.error("Cannot publish, some github commit hooks are not in a successful state.", :statuses => unsuccessful_statuses)
-            next
+        travis_check_run = nil
+        response = Octokit.check_runs_for_ref(project.path, workdir_sha1, filter: 'latest') # status: 'completed'
+        if response.total_count > 0
+          # expect only one, even if check re-run, due `filter: 'latest'`
+          if travis_check_run = response.check_runs.find { |check_run| check_run.details_url.index('travis-ci.com') }
+            if travis_check_run.status != 'completed'
+              logger.error("Cannot publish - travis-ci check hasn't completed yet", status: travis_check_run.status)
+              travis_check_run = false
+            elsif travis_check_run.conclusion != 'success'
+              logger.error("Cannot publish - travis-ci hasn't completed with a 'success'", conclusion: travis_check_run.conclusion)
+              travis_check_run = false
+            end
```

Comment:
```suggestion
            else
              logger.error("Cannot publish - travis-ci check has an unknown state", status: travis_check_run.status, conclusion: travis_check_run.conclusion)
            end
```

Reply:
that won't work since for the found `travis_check_run = response.check_runs.find` we need to assert conditions are met: `travis_check_run.status == 'completed'` (built isn't still running) and `travis_check_run.conclusion == 'success'` (run isn't in a :red_circle: state). 

`conclusion == 'success'` might be enough - anyhow having all checks explicitly should be the safer option...

Reply:
so we can just mark this as a failed check, and set `travis_check_run = false`?

Reply:
yeah we can do so - thought a separate message about "built still running" makes sense but I am fine with just logging `status:` and `conclusion:` at once - however you need to be aware what each means to understand why publish failed.

Reply:
right. the whole intent of this edit was to not leave the if/elsif claus without an `else`. happy to accept any other suggestion to handle this scenario.

Reply:
okay, we can borrow one from the code bellow :stuck_out_tongue: